### PR TITLE
Add the support for `includes` and `excludes` in unarchiving

### PR DESCRIPTION
**Pull request by smaarn**
The objective here is to be able to select only a specific set of files when unpacking using two mechanisms:

1. Whitelisting (`includes` )
2. Backlisting (`excludes`)

Integration tests were added but no unit tests.

### CODE REVIEW

File: src/main/java/com/googlecode/download/maven/plugin/internal/WGetMojo.java [hunk 1]
```diff
@@ -280,6 +282,26 @@ public class WGetMojo extends AbstractMojo {
...
     private boolean preemptiveAuth;
 
+    /**
+     * Files to include when unpacking.
+     * 
+     * <p>If left empty all files will be eligible to be unpacked.</p>
+     * 
+     * @since 1.9.0
+     */
+    @Parameter
```
Comment (longtimeago) [inline src/main/java/com/googlecode/download/maven/plugin/internal/WGetMojo.java:292]:
```suggestion
    @Parameter(property = "download.unpack.includes")
```

File: src/main/java/com/googlecode/download/maven/plugin/internal/WGetMojo.java [hunk 2]
```diff
@@ -280,6 +282,26 @@ public class WGetMojo extends AbstractMojo {
...
+    private String[] includes;
+
+    /**
+     * Files to ignore from unpacking.
+     *
+     * <p>If left empty no file will be excluded when unpacking</p>
+     *
+     * @since 1.9.0
+     */
+    @Parameter
```
Comment (longtimeago) [inline src/main/java/com/googlecode/download/maven/plugin/internal/WGetMojo.java:302]:
```suggestion
    @Parameter(property = "download.unpack.excludes")
```

File: src/it/GetAndUnpackWithIncludes/pom.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                        <phase>generate-resources</phase>
+                        <goals>
+                            <goal>wget</goal>
+                        </goals>
+                        <configuration>
+                            <!-- downloading a locally hosted jar archive, containing a "Hello, world!" text -->
+                            <url>@mrm.repository.url@/localhost/hello/1.0/hello-1.0.jar</url>
+                            <unpack>true</unpack>
+                            <includes>
+                                <include>hello.*</include>
```
Comment (longtimeago) [inline src/it/GetAndUnpackWithIncludes/pom.xml:29]: @smaarn would be super nice to have 3 files in jar/zip
- to_be_included_1
- to_be_included_2
- to_be_not_included

File: src/it/GetAndUnpackWithExcludes/pom.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                        <phase>generate-resources</phase>
+                        <goals>
+                            <goal>wget</goal>
+                        </goals>
+                        <configuration>
+                            <!-- downloading a locally hosted jar archive, containing a "Hello, world!" text -->
+                            <url>@mrm.repository.url@/localhost/hello/1.0/hello-1.0.jar</url>
+                            <unpack>true</unpack>
+                            <excludes>
+                                <exclude>hello.*</exclude>
```
Comment (longtimeago) [inline src/it/GetAndUnpackWithExcludes/pom.xml:29]: @smaarn would be super nice to have 3 files in jar/zip
- to_be_excluded_1
- to_be_excluded_2
- to_be_not_excluded

**Review by longtimeago — COMMENTED** [review]
@smaarn thank you very much for the valuable contribution!

Please, see several suggestions below :)

File: src/it/GetAndUnpackWithIncludesExcludes/pom.xml
[the remark this answers is not in the document]
Comment (smaarn) [reply src/it/GetAndUnpackWithIncludesExcludes/pom.xml:1]: Indeed. Fixed it.

File: src/it/GetAndUnpackWithIncludes/pom.xml (hunk 3 above)
Comment (smaarn) [reply src/it/GetAndUnpackWithIncludes/pom.xml]: Indeed. Reworked the reference resource to actually also include directories.

File: src/it/GetAndUnpackWithExcludes/pom.xml (hunk 4 above)
Comment (smaarn) [reply src/it/GetAndUnpackWithExcludes/pom.xml]: Reworked it to use a single reference jar shared by all include / exclude scenario.

**Pull request merged**